client/lib/signup/site-type.js

```diff
@@ -0,0 +1,59 @@
+/** @format **/
+/**
+ * Exernal dependencies
+ */
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+
+/**
+ * Current list of verticals that have/will have landing pages.
+ * A user who comes in via a landing page will not see the Site Topic dropdown.
+ */
+export const allSiteTypes = [
```

Comment:
Would this be better as an object:

```
allSiteTypes = {
  'blogger': {
    label: i18n.translate( 'Blogger' ),
    description: i18n.translate( 'Share a collection of posts.' ),
  },
  ...
}
```

That way, we probably wouldn't need the `dasherize()` method, can be more explicit with our translation strings, and can just refer to the key for `elem.type` in the component.

I don't have a particularly strong feeling about it - was just wondering what your thoughts are.

Reply:
@michaeldcain I'm not sure if we can eliminate `dasherize` function. 
`this.state.siteType` could have values either with dashes or without dashes. If siteType picks its value from the URL query parameter, then it's value is with dashes(/start?site_type=non-profit-organization), however if siteType picks its value from the selected radio option, then the value is without dashes(i.e non-sanitized original value `non-profit organization` which will be the value sent to the backend since we don't want to send a dasherized value to the backend). 
So we will need to use the dasherize function given that we will have to sanitize values for comparison. It's possible to eliminate dasherize and proceed to use the Objects method you suggested if the `site_type` query param can have spaces too. So, the vertical landing page will link to /start?vertical=yyy&site_type=non-profit+organization. Technically we can have spaces in the query param but as a matter of style, would like to check with you if we could have spaces in the query param values.